src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/InlineCompletionHandler.cs

```diff
@@ -0,0 +1,176 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.VisualStudio.LanguageServer.ContainedLanguage;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.LanguageServerClient.Razor.Extensions;
+using Microsoft.VisualStudio.LanguageServerClient.Razor.Logging;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp
+{
+    [Shared]
+    [ExportLspMethod(VSInternalMethods.TextDocumentInlineCompletionName)]
+    internal class InlineCompletionHandler : IRequestHandler<VSInternalInlineCompletionRequest, VSInternalInlineCompletionList?>
+    {
+        internal static readonly ImmutableHashSet<string> CSharpKeywords = ImmutableHashSet.Create(
```

Comment:
lmao for a few of these I went into a C# file and typed them to see what was up. Had no clue what many of em did.

Digging through these does bring up an interesting question on if we want to truncate the list of available snippet keywords so only ones that apply to Razor are available. For instance I could imagine us excluding

- `mbox`
- `namespace`
- `sim`
- `svm`

What's your view here? Not sure how excluding some might interact with bits in the completion list but I don't have the entire end-to-end in my head.

Also kind of curious how many of these plan to be revived as part of the semantic snippet work we're doing @jmarolf / @akhera99 is that known yet?

Reply:
Excluding some will have slightly weird behavior - the snippet will show up in the list but not be expandable.  I would lean towards just leaving them - semantic snippets should be the proper solution here.

And custom snippets would have the same issue.

Reply:
This is called out in this section of the spec: https://github.com/dotnet/roslyn/blob/feature/semantic-snippets/docs/ide/specs/semantic_snippets.md#scenario-2-snippets-only-show-up-where-they-are-valid

Reply:
The intent is that something like `mbox` would only show up if the required types are present and its in the right context (i.e. inside a method or top level statement). We'll need to decide what valid means for razor contexts but I assume its similar.